src/components/primitive/button/button.component.tsx

```diff
@@ -1,10 +1,47 @@
-import { ButtonProps } from './button.component.props';
-import { StyledButton } from './button.component.styles';
+import styled from 'styled-components';
+import { IconSvg } from '../../primitive';
+import { Primary, Small, GhostButton, IconWrapper, ButtonText } from './button.component.styles';
+import { ButtonComponentProps, Variant } from './button.component.props';
+
+export const Button: React.FunctionComponent<ButtonComponentProps> = (props: ButtonComponentProps) => {
+  const { variant, label, icon, onClick, ...rest } = props;
+
+  const getVariant = (variant: Variant = Variant.primary) => {
+    let ButtonComponent;
+    switch (variant) {
+      case Variant.primary:
+        ButtonComponent = Primary;
+        break;
+      case Variant.small:
+        ButtonComponent = Small;
+        break;
+      case Variant.ghost:
+        ButtonComponent = GhostButton;
+        break;
+      default:
+        ButtonComponent = Primary;
+    }
+    return ButtonComponent;
+  };
+
+  const StyledButton = getVariant(Variant[variant]);
+
+  /**
+   * custom styling goes here.
+   */
+  const StyledButtonComponent = styled(StyledButton)`
+    background-color: ${({ color, theme: { colors } }) => color && colors[color]} !important;
+  `;
 
-export const Button: React.FC<ButtonProps> = ({ children, variant, wide }) => {
   return (
-    <StyledButton variant={variant} wide={wide}>
-      {children}
-    </StyledButton>
+    // @ts-ignore - no overload matched this call
```

Comment:
Try adding BoxProps type to StyledButtonComponent to see if this error goes away.